gitlab/clients/httpx_client.py

```diff
@@ -0,0 +1,11 @@
+from typing import Optional
```

Comment:
Not sure if we need separate modules at this point as they're quite small.

Reply:
The intention is moving the functionality to the clients classes.  This PR will lay the ground for the coming changes.

Reply:
Ah ok :) I'd personally go for `httpx.py` and `requests.py` as we already have the namespace.

Reply:
I like the file name to be aligned with the class name